datalad_next/constraints/basic.py

```diff
@@ -283,6 +283,39 @@ def short_description(self):
         return '%s:{%s}' % (self._key, ', '.join([repr(c) for c in self._allowed]))
 
 
+class EnsureSet(Constraint):
```

Comment:
I am thinking about a good naming rule for this since #277. Presently I think, the name should reflect the outcome of the processing. Here we get `input is output` whenever a `required.issubset()` criterion is fulfilled. What about something like `EnsureContainsSet()`?

This should cause no expectations of type-changes.

Docs should state that valid input is any iterable.